File: examples/python/log.py
"""
`log` global prints messages to the Output and Errors tab in Extensions -> Burp Scripting
print() sends messages there too.

[logging.py][INFO] - GET - https://www.ivision.com/
[logging.py][INFO] - Logs INFO
[logging.py][DEBUG] - Got a response
[logging.py][INFO] - {'key': 'value'}
[logging.py][ERROR] - oh no
ZeroDivisionError: division by zero
"""

# Enable log.debug()
log.enableDebug(True)

def on_request(req):
    log.info(f"{req.method()} - {req.url()}")
    print("Logs INFO")
    return req


def on_response(res):
    log.debug("Got a response")
    obj = res.bodyToJson()
    print(obj)
    try:
        x = 1 / 0
    except Exception as ex:
        # Logs to Errors tab of the extension
        log.error("oh no", ex)
    return res

File: examples/js/log.mjs
/**
 * `log` global prints messages to the Output and Errors tab in Extensions -> Burp Scripting
 * console.log, console.error, console.warn, console.info are also available
 *
 * [logging.mjs][INFO] - GET - https://www.ivision.com/
 * [logging.mjs][INFO] - Logs INFO
 * [logging.mjs][DEBUG] - Got a response
 * [logging.mjs][INFO] - {'key': 'value'}
 * [logging.mjs][ERROR] - oh no
 * TypeError: Cannot set property 'foo' of null
 */

export function onRequest(req) {
    log.info(`${req.method()} - ${req.url()}`)
    console.log("Logs INFO")
    return req
}

export function onResponse(res) {
    log.info(`${res.statusCode()} - ${res.reasonPhrase()}`);
    const obj = res.bodyToJson();
    console.log(obj);
    try {
        const obj = null;
        obj.foo = "bar";
    } catch (error) {
        log.error("oh no", error);
    }
    return res;
}
